lib/routemap.c

```diff
@@ -1616,19 +1660,30 @@ route_map_result_t route_map_apply(struct route_map *map,
 					}
 					if (next == NULL) {
 						/* No clauses match! */
-						return ret;
+						goto route_map_apply_end;
 					}
 				}
 				}
 			} else if (index->type == RMAP_DENY)
 			/* 'deny' */
 			{
-				return RMAP_DENYMATCH;
+				ret = RMAP_DENYMATCH;
+				goto route_map_apply_end;
 			}
 		}
 	}
 	/* Finally route-map does not match at all. */
-	return RMAP_DENYMATCH;
+	ret = RMAP_DENYMATCH;
+
+route_map_apply_end:
+	if (rmap_debug) {
+		zlog_debug("Route-map: %s, prefix: %s, result: %s",
+			   (map ? map->name : "null"),
+			   prefix2str(prefix, buf, sizeof(buf)),
```

Comment:
if I'm not mistaken we don't need `buf` and `buf2`, we can just use one `buf` right?